staging/src/k8s.io/kubectl/pkg/cmd/version/skew_warning.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"io"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"math"
+	"regexp"
+	"strconv"
+)
+
+// supportedMinorVersionSkew is the maximum supported difference between the client and server minor versions.
+// For example: client versions 1.18, 1.19, and 1.20 would be within the supported version skew for server version 1.19,
+// and server versions 1.18, 1.19, and 1.20 would be within the supported version skew for client version 1.19.
+const supportedMinorVersionSkew = 1
+
+func printVersionSkewWarning(w io.Writer, clientVersion, serverVersion apimachineryversion.Info) error {
+	if supported, err := isWithinSupportedVersionSkew(clientVersion, serverVersion); err != nil {
+		return err
+	} else if !supported {
+		fmt.Fprintf(w, "WARNING: version difference between client (%s.%s) and server (%s.%s) exceeds the supported minor version skew of +/-%d\n",
```

Comment:
Yeah, I don’t really know what else to say here. It’s really not horrible to use an out of skew client, but if you are running into problems, you should be aware that it could be an unsupported skew.

A main reason for adding this warning is to make people aware of the unsupported skew before they open an issue in the kubectl repo.

but you’re right, it is not super actionable.

Reply:
Hey @briandorsey I believe the warning you created is super valid but it's difficult to avoid users to open issues with unsupported skew IMO. 

Probably we might have such scenarios with users out there:
  - didn't understand the message - what's skew +/- 1? 
  - simple ignore it as they don't know what to do next 
  - not sure "how to fix it"
  - not sure "how to make the cluster in a minimum good shape" to open the issue. 
   
From my side, the code looks great but I would request more people to review the message to have more ideas. Probably a good PR for request ideas in the SIG-CLI meeting.